timetracker/entries/views.py

```diff
@@ -1,113 +1,43 @@
 from django.shortcuts import render, get_object_or_404, redirect
+from django.views.generic import DetailView, ListView, CreateView, RedirectView
+from django.core.urlresolvers import reverse_lazy
 
 from .forms import EntryForm, ProjectForm, ClientForm
 from .models import Client, Entry, Project
 
+class RootRedirectView(RedirectView):
+    permanent = False
 
-def clients(request):
-    if request.method == 'POST':
-        # Create our form object with our POST data
-        form = ClientForm(request.POST)
-        if form.is_valid():
-            # If the form is valid, create a client with submitted data
-            # Below is the shortcut equivalent of:
-            # client = Client()
-            # client.name = form.cleaned_data['name']
-            # client.save()
-            # Sometimes you don't want to save the object until the end,
-            # sometimes you don't care!
-            client = Client.objects.create(name=form.cleaned_data['name'])
-            return redirect('client-list')
-    else:
-        form = ClientForm()
+    def get_redirect_url(self, *args, **kwargs):
+        return reverse_lazy('client-list')
 
-    client_list = Client.objects.all()
-    return render(request, 'clients.html', {
-        'client_list': client_list,
-        'form': form,
-    })
+class ClientDetailView(DetailView):
+    model = Client
+    template_name = 'client_detail.html'
 
+class ClientListView(ListView):
+    model = Client
+    template_name = 'client_list.html'
```

Comment:
`client_list.html` should be the automatic default for this view. No need to specify `template_name` explicitly.